src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -3247,6 +3265,20 @@ uct_ib_mlx5dv_check_ddp(struct ibv_context *ctx, uct_ib_mlx5_md_t *md)
     return UCS_OK;
 }
 
+static void uct_ib_mlx5dv_md_check_odp(uct_ib_mlx5_md_t *md)
+{
+    const char *device_name = uct_ib_device_name(&md->super.dev);
+    const char *reason;
+
+    if (!uct_ib_mlx5_md_check_odp_common(md, &reason)) {
+        ucs_debug("%s: ODP is disabled because %s", device_name, reason);
+        return;
+    }
+
+    md->super.reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
```

Comment:
should we use `md_config->ext.odp.mem_types;` here?

Reply:
right, it was existing bug that didn't consider mem_types in non-DevX case, force-pushed the fix